lib/hooks/useTokenHolder.js

```diff
@@ -54,33 +52,75 @@ function useFetchTokenHolder (address, blockNumber) {
 }
 
 async function getTokenHolder (address, chainId, blockNumber) {
-  const query = tokenHolderQuery(blockNumber)
-
-  const variables = { id: address.toLowerCase() }
-
   try {
+    const query = tokenHolderQuery(blockNumber)
+    const variables = { id: address.toLowerCase() }
     const subgraphData = await request(
       POOLTOGETHER_CURRENT_GOVERNANCE_GRAPH_URIS[chainId],
       query,
       variables
     )
 
-    if (!subgraphData.tokenHolder) {
-      return null
+    if (!subgraphData.tokenHolder && !subgraphData.delegate) {
+      return {
+        tokenBalance: '0',
+        hasDelegated: false,
+        selfDelegated: false,
+        hasBalance: false,
+        canVote: false,
+        delegateAddress: '',
+        votes: '0',
+        tokenHoldersRepresentedAmount: 0
+      }
     }
 
+    const hasDelegated =
+      Boolean(subgraphData?.tokenHolder?.delegate?.id) &&
+      subgraphData?.tokenHolder?.delegate?.id !== ethers.constants.AddressZero
+    const selfDelegated =
+      address.toLowerCase() === subgraphData?.tokenHolder?.delegate?.id?.toLowerCase()
+    const hasBalance = Number(subgraphData?.tokenHolder?.tokenBalance) > 0
+    const canVote = selfDelegated || (!hasDelegated && subgraphData.delegate !== '0')
+    const delegateAddress = hasDelegated
```

Comment:
I would like to vote for not using nested ternary statements, I find them really hard to discern at a glance. I do like the shallow ternary, but nested just feels messy. Maybe we can break some of this logic out into separate helper functions?